Breaking change: getAI() parameter type incompatible with AIClientOptions

## Bug Description

The `getAI()` function parameter type changed, making it incompatible with the `AIClientOptions` type that SMRT classes use, breaking existing code.

## TypeScript Error

```typescript
src/meetings.ts:566:28 - error TS2345: Argument of type 'AIClientOptions | AIClient' 
is not assignable to parameter of type 'GetAIOptions | undefined'.
  Type 'AIClientOptions' is not assignable to type 'GetAIOptions | undefined'.
    Type 'AIClientOptions' is not assignable to type 'OpenAIOptions'.
      Types of property 'type' are incompatible.
        Type 'string | undefined' is not assignable to type '"openai" | undefined'.
          Type 'string' is not assignable to type '"openai"'.

566     const ai = await getAI(this.options.ai);
                           ~~~~~~~~~~~~~~~
```

## Problem

SMRT classes (like `Meetings`) accept `AIClientOptions` in their constructor:
```typescript
interface AIClientOptions {
  type?: string;  // Can be 'openai', 'anthropic', 'google', etc.
  apiKey?: string;
  // ...
}
```

But `getAI()` now expects `GetAIOptions`:
```typescript
type GetAIOptions = OpenAIOptions | AnthropicOptions | GoogleOptions | ...

interface OpenAIOptions {
  type: "openai";  // ← Literal type, not string
  apiKey: string;
  // ...
}
```

## Breaking Change

This is a **breaking API change** introduced in recent SDK refactoring.

**Before:**
```typescript
const meetings = new Meetings({
  ai: { type: 'openai', apiKey: process.env.OPENAI_API_KEY }
});
// meetings internally calls getAI(this.options.ai) - worked fine
```

**After:**
Type error because `type: string` doesn't satisfy `type: "openai"` literal type requirement.

## Impact

- **High** - Breaks all SMRT classes that use `getAI()` internally
- Affects: `Meetings`, `Praeco`, and any class accepting AI options
- Multiple compilation errors in praeco:
  - src/meetings.ts:566
  - src/meetings.ts:666
  - src/praeco.ts:544

## Possible Solutions

### Option 1: Widen GetAIOptions to accept AIClientOptions
```typescript
export type GetAIOptions = AIClientOptions | OpenAIOptions | AnthropicOptions | ...
```

### Option 2: Add type guard/converter
```typescript
function toGetAIOptions(options: AIClientOptions): GetAIOptions {
  if (options.type === 'openai') return options as OpenAIOptions;
  // ...
}
```

### Option 3: Make literal types compatible
```typescript
interface AIClientOptions {
  type?: 'openai' | 'anthropic' | 'google' | 'aws';  // ← Literal union
}
```

## Environment

- @happyvertical/ai: workspace:* (latest)
- @happyvertical/smrt-core: workspace:* (latest)
- TypeScript: 5.9.3

## Related

- Affects praeco: happyvertical/praeco#15

--- a/packages/ai/src/shared/factory.ts
+++ b/packages/ai/src/shared/factory.ts
@@ -5,6 +5,7 @@
 
 import { loadEnvConfig, ValidationError } from '@happyvertical/utils';
 
+import type { AIClientOptions } from './client';
 import type {
   AIInterface,
   AnthropicOptions,
@@ -25,7 +26,9 @@ import type {
  * @param options - The AI provider options to check
  * @returns True if options are for OpenAI provider (including default case)
  */
-function isOpenAIOptions(options: GetAIOptions): options is OpenAIOptions {
+function isOpenAIOptions(
+  options: GetAIOptions | AIClientOptions,
+): options is OpenAIOptions {
   return !options.type || options.type === 'openai';
 }
 
@@ -34,7 +37,9 @@ function isOpenAIOptions(options: GetAIOptions): options is OpenAIOptions {
  * @param options - The AI provider options to check
  * @returns True if options are for Gemini provider
  */
-function isGeminiOptions(options: GetAIOptions): options is GeminiOptions {
+function isGeminiOptions(
+  options: GetAIOptions | AIClientOptions,
+): options is GeminiOptions {
   return options.type === 'gemini';
 }
 
@@ -44,7 +49,7 @@ function isGeminiOptions(options: GetAIOptions): options is GeminiOptions {
  * @returns True if options are for Anthropic provider
  */
 function isAnthropicOptions(
-  options: GetAIOptions,
+  options: GetAIOptions | AIClientOptions,
 ): options is AnthropicOptions {
   return options.type === 'anthropic';
 }
@@ -55,7 +60,7 @@ function isAnthropicOptions(
  * @returns True if options are for Hugging Face provider
  */
 function isHuggingFaceOptions(
-  options: GetAIOptions,
+  options: GetAIOptions | AIClientOptions,
 ): options is HuggingFaceOptions {
   return options.type === 'huggingface';
 }
@@ -65,7 +70,9 @@ function isHuggingFaceOptions(
  * @param options - The AI provider options to check
  * @returns True if options are for Bedrock provider
  */
-function isBedrockOptions(options: GetAIOptions): options is BedrockOptions {
+function isBedrockOptions(
+  options: GetAIOptions | AIClientOptions,
+): options is BedrockOptions {
   return options.type === 'bedrock';
 }
 
@@ -75,7 +82,7 @@ function isBedrockOptions(options: GetAIOptions): options is BedrockOptions {
  * @returns True if options are for Claude CLI provider
  */
 function isClaudeCliOptions(
-  options: GetAIOptions,
+  options: GetAIOptions | AIClientOptions,
 ): options is ClaudeCliOptions {
   return options.type === 'claude-cli';
 }
@@ -94,7 +101,11 @@ function isClaudeCliOptions(
  *
  * User-provided options always take precedence over environment variables.
  *
- * @param options - Configuration options for the AI provider. Must include provider type and credentials.
+ * Accepts both GetAIOptions (provider-specific options with literal types)
+ * and AIClientOptions (legacy interface with generic string type) for
+ * backward compatibility with existing code.
+ *
+ * @param options - Configuration options for the AI provider. Can be GetAIOptions or AIClientOptions.
  * @returns Promise resolving to an AI provider instance that implements the AIInterface
  * @throws {ValidationError} When the provider type is unsupported or invalid
  *
@@ -116,9 +127,15 @@ function isClaudeCliOptions(
  *   apiKey: process.env.ANTHROPIC_API_KEY!,
  *   defaultModel: 'claude-3-5-sonnet-20241022'
  * });
+ *
+ * // Works with AIClientOptions (legacy interface)
+ * const clientOptions: AIClientOptions = { type: 'openai', apiKey: '...' };
+ * const legacy = await getAI(clientOptions);
  * ```
  */
-export async function getAI(options: GetAIOptions = {}): Promise<AIInterface> {
+export async function getAI(
+  options: GetAIOptions | AIClientOptions = {},
+): Promise<AIInterface> {
   // Load environment variables with user options taking precedence
   options = loadEnvConfig(options as Record<string, any>, {
     packageName: 'ai',

